superset-frontend/src/SqlLab/components/AceEditorWrapper/index.tsx

```diff
@@ -98,14 +100,27 @@ const AceEditorWrapper = ({
     'dbId',
     'sql',
     'functionNames',
-    'schemaOptions',
     'tableOptions',
     'validationResult',
     'schema',
   ]);
+  const { data: schemaOptions } = useSchemas({ dbId: queryEditor.dbId });
   const currentSql = queryEditor.sql ?? '';
   const functionNames = queryEditor.functionNames ?? [];
-  const schemas = queryEditor.schemaOptions ?? [];
+
+  // Loading schema, table and column names as auto-completable words
+  const { schemas, schemaWords } = useMemo(
+    () => ({
+      schemas: schemaOptions ?? [],
+      schemaWords: (schemaOptions ?? []).map(s => ({
+        name: s.label,
+        value: s.value,
+        score: SCHEMA_AUTOCOMPLETE_SCORE,
+        meta: 'schema',
+      })),
+    }),
+    [schemaOptions],
+  );
```

Comment:
Do we really need a `useMemo` here? It may look cleaner to keep the words logic with other autocomplete words.

Reply:
Yes. useMemo will reduce the cost of `schemaWords` mapping from schemaOptions.
(For example, airbnb has > 10,000 schema options so `useMemo` should be needed)